tdrs-backend/tdpservice/users/test/test_auth.py

```diff
@@ -86,6 +88,302 @@ def test_login_fails_without_state(api_client):
     response = api_client.get("/v1/login", {"code": "dummy"})
     assert response.status_code == status.HTTP_302_FOUND
 
+@pytest.mark.django_db
+def test_login_with_valid_state_and_code(mocker, api_client):
+    """Test login with state and code."""
+    os.environ["JWT_KEY"] = test_private_key
+    nonce = "testnonce"
+    state = "teststate"
+    code = secrets.token_hex(32)
+    mock_post = mocker.patch("tdpservice.users.api.login.requests.post")
+    token = {
+        "access_token": "hhJES3wcgjI55jzjBvZpNQ",
+        "token_type": "Bearer",
+        "expires_in": 3600,
+        "id_token": os.environ["MOCK_TOKEN"]
+    }
+    mock_decode = mocker.patch("tdpservice.users.api.login.jwt.decode")
+    decoded_token = {
+        "email": "test@example.com",
+        "email_verified": True,
+        "nonce": nonce,
+        "iss": "https://idp.int.identitysandbox.gov",
+        "sub": "b2d2d115-1d7e-4579-b9d6-f8e84f4f56ca",
+        "verified_at": 1577854800
```

Comment:
Whats the specific reason for picking this epoch stamp that converts to `Wednesday, January 1, 2020`. Adding a comment as to why we're picking this epoch stamp might help.

Reply:
No particular reason for this. It was just an example token that we grabbed to mock with. It doesn't make any difference for the mock.

Reply:
copy does this mean it'll break  on new years? Would it be feasible to have this set to lets ays 15min from the current system time?

Reply:
No, it doesn't actually read the value